kore/src/Kore/Step/Step.hs

```diff
@@ -139,16 +140,45 @@ unwrapRule = Rule.mapVariables Target.unwrapVariable
 
 {- | Remove axiom variables from the substitution and unwrap all variables.
  -}
-unwrapConfiguration
-    :: Ord variable
+unwrapAndQuantifyConfiguration
+    :: forall variable
+    .   ( Ord variable
+        , Show variable
+        , SortedVariable variable
+        , Unparse variable
+        )
     => Pattern (Target variable)
     -> Pattern variable
-unwrapConfiguration config@Conditional { substitution } =
-    Pattern.mapVariables Target.unwrapVariable
-        config { Pattern.substitution = substitution' }
+unwrapAndQuantifyConfiguration config@Conditional { substitution } =
+    if List.null targetVariables
+    then unwrappedConfig
+    else
+        Pattern.fromTermLike
+            (List.foldl'
+                quantify
+                (Pattern.toTermLike unwrappedConfig)
```

Comment:
Doesn't this discard the predicate and substitution?

Reply:
No, it merges them into the term.